ui/src/lib/store/canvasSlice.tsx

```diff
@@ -46,31 +49,55 @@ const level2color = {
 };
 
 /**
- * The temporary reactflow nodes for paste/cut.
+ * Creare the temporary nodes as well as the temporary pods based on the given pod.
  * @param pod
  * @param position
+ * @param parent
+ * @param level
  * @returns
  */
-function createTemporaryNode(pod, position) {
+function createTemporaryNode(pod, position, parent = "ROOT", level = 0): any {
```

Comment:
One bug: after copying a scope, the nodes inside the scope can be moved out. The reason is that the `parent: extent` field is not set for the new scope.

For reference, this `parent: extent` field is set when a pod is moved into a scope.

https://github.com/codepod-io/codepod/blob/5b39efe1df7bcdf62c53b2d264a6e51011148451/ui/src/lib/store/canvasSlice.tsx#L608-L617

Reply:
fixed

Reply:
I see. This may be the same reason why a pod can be moved out after refreshing the page. Let me take a look and fix the bug tonight.

Reply:
> fixed

@li-xin-yi